src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -271,6 +274,87 @@ public int read(byte b[], int off, int len) throws IOException {
         return readBytes(b, off, len);
     }
 
+    public byte[] readAllBytes() throws IOException {
+        long length = length();
+        int capacity;
+        if (length != 0L) {
+            long position = position();
+            long size = length - position;
+            if (size > (long)Integer.MAX_VALUE)
+                throw new OutOfMemoryError("Required array size too large");
```

Comment:
What do you think of adding "length, position, and size" to the exception message?

Reply:
We usually don't do that for OutOfMemoryError because concatenation implies allocation

Reply:
Well, here `OutOfMemory` is more like "I can't read all the data since the array won't fit".
It would definitely help to have actual file size there, especially in case filesystem returns weird values (e.g. negative file length).

This branch will almost never be taken, and even if taken, it won't be "low memory" condition. So the allocation does not hurt, however, it would simplify the analysis should the case trigger.

Reply:
What about adding the filename into exception message?